scanner/indexer/indexer.go

```diff
@@ -377,7 +393,15 @@ func createManifestDigest(hashID string) (claircore.Digest, error) {
 // a list of layers.
 func getContainerImageLayers(ctx context.Context, ref name.Reference, o options) ([]v1.Layer, error) {
 	// TODO Check for non-retryable errors (permission denied, etc.) to report properly.
-	desc, err := remote.Get(ref, remote.WithContext(ctx), remote.WithAuth(o.auth), remote.WithPlatform(o.platform))
+	client := &http.Client{
+		Transport: remoteTransport,
+	}
+	if o.insecure {
+		client = &http.Client{
+			Transport: insecureRemoteTransport,
+		}
+	}
+	desc, err := remote.Get(ref, remote.WithContext(ctx), remote.WithAuth(o.auth), remote.WithPlatform(o.platform), remote.WithTransport(client.Transport))
```

Comment:
no need to create the `client`. You can just use `remoteTransport` or `insecureRemoteTransport`, directly